lua/nvim-navic/init.lua

```diff
@@ -117,60 +198,44 @@ local function in_range(cursor_pos, range)
 	return 0
 end
 
-local function update_context(for_buf)
```

Comment:
Why are change this function?

Reply:
Just for simplicity. It took me a while to figure out how the original function was working.  This implementation seemed like it would be easier to follow -- recursion tends to work well for tree walking.

Reply:
My original reason for looking at that function was because I was running into cases where the context wasn't being updated. Like, if I started at the top of a file and stepped down line-by-line, the context would generally update to the expected value, but if I started at the bottom and stepped up, the displayed context wasn't always correct.